hatch_pip_compile/plugin.py

```diff
@@ -347,3 +371,37 @@ def pipools_environment_dict(self) -> Dict[str, Any]:
         Get the environment dictionary
         """
         return self.metadata.hatch.config.get("envs", {})
+
+    def _piptools_install_dependencies(self) -> None:
+        """
+        Install dependencies with `pip-sync` or `pip`-install`
+        """
+        if self.install_method == "pip" and self.dependencies:
+            self._pip_install_cli()
+        elif self.install_method == "pip-sync":
+            self._pip_sync_cli()
+        else:
+            msg = f"Invalid pip-tools install method: {self.install_method}"
+            raise NotImplementedError(msg)
+
+    def install_project_no_deps(self) -> None:
+        """
+        Install the project (`--no-deps`)
+        """
+        with self.safe_activation():
+            proj_with_feats = self.apply_features(str(self.root))
```

Comment:
What does it mean to install the project with extras (`apply_features`) but then not actually install any dependencies? Surely the extras are ignored then.

Reply:
Oh, that's a great point. I just copied the upstream implementation but didn't think of that.

Reply:
So I think `self.apply_features` can just be dropped without consequence